src/app/_components/BlogSearchInput.tsx

```diff
@@ -2,30 +2,48 @@
 
 import { useState } from 'react'
 
+import { useSearchParams } from 'next/navigation'
+
 import { useDebounceCallback } from 'usehooks-ts'
 
+import { useSearch } from '@/hooks/useSearch'
 import { useUpdateSearchParams } from '@/hooks/useUpdateSearchParams'
 
 import { SearchInput } from '@/components/SearchInput'
 
+import { PAGE_KEY, SEARCH_KEY } from '@/constants/searchParams'
+
 type BlogSearchInputProps = {
-  searchQuery: string
+  query: ReturnType<typeof useSearch>['searchQuery']
 }
 
-export function BlogSearchInput({ searchQuery }: BlogSearchInputProps) {
-  const [value, setValue] = useState(searchQuery)
+const TYPE_DEBOUNCE_DELAY = 400
```

Comment:
Why is this prefixed with `TYPE` - What does that mean in this context?

Reply:
It's type as in typing on a keyboard, but I'm realising it's not exactly right. Maybe `DEBOUNCE_DELAY` alone would be better?

Reply:
Agree! `DEBOUNCE_DELAY` sounds good to me.

Reply:
✅